include/ze_info/client.hpp

```diff
@@ -47,7 +47,7 @@ class client
 
 				// read the elements in the file into a vector  
 				int n = 0;
-				while (inputFile >> value) {
+				while (inputFile >> value && n < queries) {
```

Comment:
Why do we need this additional condition? for a case when someone want to use dist from run with different number of queries?

Reply:
right. currently you can get error connected to vector length

Reply:
hmm, IMHO it should be done in that way:
if size doesn't match. delete old, create new file - prompt that size did match and new one has been created

Reply:
I've just pushed the solution discussed offline